### PR TITLE
Allow PBOG to nudge blocking helicopters.

### LINKED ISSUE #16798: Landed helicopters do not move when you try to build on them
When you try to place a building on the land units they move to open space for the building, aircraft do not do that.

### PR DESCRIPTION
**Pull request by pchote**
Fixes #16798.

### CODE REVIEW

**Review by abcdefg30 — DISMISSED** [review]

File: OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs [hunk 1]
```diff
@@ -306,15 +306,26 @@ IEnumerable<Order> ClearBlockersOrders(World world, CPos topLeft)
...
 
 			foreach (var blocker in blockers.Where(x => x.Trait != null))
 			{
-				var availableCells = neightborTiles.Where(t => blocker.Trait.CanEnterCell(t)).ToList();
-				if (availableCells.Count == 0)
+				CPos moveCell;
+
+				if (blocker.Trait is Mobile)
+				{
+					var availableCells = neightborTiles.Where(t => ((Mobile)blocker.Trait).CanEnterCell(t)).ToList();
```
[the remark this answers is not in the document]
Comment (teinarss) [reply OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs]: Also a spelling mistake on `neightborTiles` not from this PR tho, but cant unseen it now :(

File: OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs [hunk 2]
```diff
@@ -306,15 +306,26 @@ IEnumerable<Order> ClearBlockersOrders(World world, CPos topLeft)
 
 			var blockers = allTiles.SelectMany(world.ActorMap.GetActorsAt)
 				.Where(a => a.Owner == queue.Actor.Owner && a.IsIdle)
-				.Select(a => new TraitPair<Mobile>(a, a.TraitOrDefault<Mobile>()));
+				.Select(a => new TraitPair<IMove>(a, a.TraitOrDefault<IMove>()));
 
 			foreach (var blocker in blockers.Where(x => x.Trait != null))
```
Comment (teinarss) [inline OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs]: Also out of this scope, but we could move this Where up to the rest of the query.

**Comment by pchote** [comment]
Fixed and rebased.

**Review by teinarss — APPROVED** [review]

**Comment by teinarss** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/316eff22b522ea88f152485538bd83e764e14277...61fda59c74a271b3209ceb6bfcc5e64e0e7d54ae

**Pull request merged**